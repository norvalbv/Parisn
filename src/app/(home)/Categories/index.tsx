'use client';

import { collections } from '@/src/__mocks__/dataApiMock';
import Loader from '@/src/components/Loading';
import PageIndicators from '@/src/components/PageIndicators';
import StyledLink from '@/src/components/StyledLink';
import { COMPANY_NAME } from '@/src/constants';
import React, { ReactElement, useState } from 'react';

const Categories = (): ReactElement => {
  // const { data } = useCollections();
  const [page, setPage] = useState(0);
  const data = collections;

  if (!data) return <Loader />;

  const pageCount = Math.ceil(data.length / 3);

  // Create a shallow copy of a data that is limited to three categories per page.
  const processedData = data.slice(page * 3, page * 3 + 3);

  return (
    <section className="my-20 overflow-hidden px-4 md:my-40 md:px-8">
      <div className="mx-auto mb-12 flex max-w-7xl flex-col items-center justify-between md:mb-16 md:flex-row">
        <div className="mb-6 text-center md:mb-0 md:text-left">
          <h3 className="mb-2 text-3xl font-semibold tracking-tight">
            <span className="logo-text-medium">{COMPANY_NAME}</span>&apos;s Collections
          </h3>
          <p className="text-xs font-medium uppercase tracking-wider text-primary-neutral/80">
            Uncover Unparalleled Elegance
          </p>
        </div>
      </div>

      <div className="relative mx-auto max-w-7xl">
        <div className="grid grid-cols-1 gap-8 md:grid-cols-3">
          {processedData.map((collection) => (
            <div
              key={collection.collections}
              className="group relative h-[24rem] overflow-hidden rounded-2xl border border-white/10 bg-gradient-to-br from-primary-light/[.03] via-primary-light/5 to-primary-light/10 backdrop-blur-sm transition-all hover:border-white/20"
            >
              <div className="relative h-full w-full">
                <img
                  src={collection.image}
                  alt={`${collection.collections} collection cover`}
                  className="h-full w-full object-cover brightness-75 transition-transform duration-500 group-hover:scale-105"
                />
                <div className="absolute inset-0 bg-gradient-to-b from-transparent via-black/20 to-black/60" />
                <div className="absolute bottom-0 left-0 right-0 space-y-4 p-6">
                  <h4 className="text-center text-xl font-semibold uppercase tracking-wide">
                    {collection.collections}
                  </h4>
                  <div className="flex justify-center">
                    <StyledLink
                      to={`/collections/${collection.collections}`}
                      text="View Collection"
                    />
                  </div>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>

      <div className="flex justify-center">
        <PageIndicators pages={pageCount} currentPage={page} setCurrentPage={setPage} />
      </div>
    </section>
  );
};

export default Categories;
