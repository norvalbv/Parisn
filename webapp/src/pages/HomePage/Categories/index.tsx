import React, { ReactElement, useState } from 'react';
import StyledLink from 'components/StyledLink';
import Radial from 'components/SVG/Design';
// import { useCollections } from 'services/DataApiService';
import Loader from 'components/Loading';
import { collections } from '__mocks__/dataApiMock';
import PageIndicators from 'components/PageIndicators';

const Categories = (): ReactElement => {
  // const { data } = useCollections();
  const [page, setPage] = useState(0);
  const data = collections;

  if (!data) return <Loader />;

  const pageCount = Math.ceil(data.length / 3);

  // Create a shallow copy of a data that is limited to three categories per page.
  const processedData = data.slice(page * 3, page * 3 + 3);

  return (
    <section className="relative flex flex-col items-center px-4 md:px-0">
      <Radial colour="purple" className="hidden md:block" />
      <p className="uppercase text-center">Uncover Unparalleled Elegance</p>
      <h3 className="mb-6 md:mb-10 text-center">
        <span className="logo-text-medium">PARISN</span>&apos;s Exceptional Collections
      </h3>
      <div className="flex flex-col md:flex-row items-center gap-6 md:gap-10">
        {processedData.map((collection) => (
          <div
            className="relative h-[24rem] md:h-[31.875rem] w-full md:w-[19.625rem] rounded"
            key={collection.collections}
          >
            <img
              src={collection.image}
              alt={`${collection.collections} collection cover`}
              className="h-full w-full object-cover rounded brightness-[.65]"
            />
            <div className="absolute bottom-0 flex h-[6rem] md:h-[8.25rem] w-full flex-col items-center justify-center gap-4 md:gap-6 bg-gradient-to-b from-primary-dark/30 via-primary-dark/20 to-primary-dark/20">
              <h4 className="font-thin leading-[1.8125rem] text-center">{collection.collections}</h4>
              <StyledLink to={`/collections/${collection.collections}`} text="View All" />
            </div>
          </div>
        ))}
      </div>
      <PageIndicators pages={pageCount} currentPage={page} setCurrentPage={setPage} className="mt-6 md:mt-10" />
    </section>
  );
};

export default Categories;
