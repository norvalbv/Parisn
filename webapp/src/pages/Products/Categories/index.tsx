import React, { ReactElement } from 'react';
import { Link } from 'react-router-dom';
import { PRODUCT_1_IMAGE } from '../../../constants';

const arr = [1, 2, 3];

const Categories = (): ReactElement => {
  return (
    <div className="grid grid-cols-3 h-screen">
      {arr.map((__, idx) => (
        <div
          key={idx}
          className="flex flex-col flex-wrap justify-center items-center pt-10"
          data-testid="Shoes"
        >
          <Link to="/catalogue" className="flex justify-center items-center">
            <img
              src={PRODUCT_1_IMAGE}
              alt={PRODUCT_1_IMAGE}
              className="w-[24rem] h-[34rem] cursor-pointer hover:scale-110 transition-all rounded-xl"
            />
          </Link>
          <p className="underline mt-10 text-xl uppercase tracking-widest font-thin underline-offset-8">
            Shoes
          </p>
        </div>
      ))}
    </div>
  );
};

export default Categories;
