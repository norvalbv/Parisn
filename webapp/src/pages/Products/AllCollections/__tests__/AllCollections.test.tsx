import renderer from 'react-test-renderer';

import Categories from '..';

describe('<Categories />', () => {
  test('Categories props with mandatory props', () => {
    const tree = renderer.create(<Categories />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
