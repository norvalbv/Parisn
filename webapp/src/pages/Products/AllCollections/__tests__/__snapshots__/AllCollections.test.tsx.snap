// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> Categories props with mandatory props 1`] = `null`;
